client/util/scroll.ts

```diff
@@ -61,42 +60,35 @@ function isAtBottom(): boolean {
 		>= document.documentElement.offsetHeight
 }
 
-// Scroll to page bottom, if previously at the bottom and the scroll away
-// happened on DOM update and not user's scroll
-function onFrame() {
-	if (page.thread) {
-		if (scrolled) {
-			checkBottom()
-			scrolled = false
-		} else if (atBottom && !isAtBottom()) {
-			window.scrollTo(0, document.body.scrollHeight)
-		}
-	}
-	requestAnimationFrame(onFrame)
-}
-
-// Firefox's multithreaded scrolling introduces a race between scroll events and
-// scrolling API updates. Just disable it for them.
-if (!isCuck) {
-	requestAnimationFrame(onFrame)
-
-	document.addEventListener("scroll", () => scrolled = true, {
-		passive: true,
-	})
+// Scrolled when the page is scrolled, unless it's at the bottom
+document.addEventListener("scroll", () => {
+	scrolled = !isAtBottom()
+	atBottom = isAtBottom()
+}, { passive: true })
 
-	// Unlock from bottom, when the tab is hidden, unless set not to
-	document.addEventListener("visibilitychange", () => {
-		const opts = trigger("getOptions")
-		if (document.hidden && (opts && !opts.alwaysLock)) {
-			atBottom = false
+// Use a MutationObserver to jump to the bottom of the page when a new
+// post is made, unless the user has scrolled up from the bottom
+let threadContainer = document.getElementById("thread-container")
+let threadObserver = new MutationObserver((mut) => {
+	mut.forEach(() => {
+		if (!scrolled) {
+			scrollToBottom()
 		}
 	})
-} else {
-	// Permanently hide locking UI
-	const el = document.createElement("style")
-	el.innerHTML = `#lock, #alwaysLock, label[for="alwaysLock"]{display: none;}`
-	document.head.append(el)
-}
+})
+threadObserver.observe(threadContainer, {
```

Comment:
This works for new post additions, but not, when an existing post is modified (for example, blockquote text mutations).